test/Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests/command/app/ShellCommand.cs

```diff
@@ -0,0 +1,143 @@
+﻿using Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests.apps.languages;
+using Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests.apps.type;
+using Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests.apps.brokers;
+using Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests.executor;
+using Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests.executor.process;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests.Util;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Kafka.LangEndToEndTests.command.app
+{
+    public class ShellCommand: Command<Process>
+    {
+        private Process process;
+        
+        private readonly Language language;
+        private readonly BrokerType brokerType;
+        private ShellCommandType shellCommandType = ShellCommandType.DOCKER_RUN;
+        private bool isNightlyBuild = false;
+        
+        private IExecutor<string, Process> processExecutor = null;
+        
+        private ShellCommand(ShellCommandBuilder shellCommandBuilder)
+        {
+            this.language = shellCommandBuilder.GetLanguage();
+            this.brokerType = shellCommandBuilder.GetBrokerType();
+            this.shellCommandType = shellCommandBuilder.GetShellCommandType();
+            this.processExecutor = new ProcessExecutor();
+        }
+
+        public async Task<Process> ExecuteCommandAsync()
+        {
+            string cmd = Constants.FUNC_START;
+            if (ShellCommandType.DOCKER_RUN == shellCommandType)
+            {
+                cmd = buildDockerStartCmd();
+            }
+            else if (ShellCommandType.DOCKER_KILL == shellCommandType)
+            {
+                cmd = buildDockerKillCmd();
+            }
+
+            this.process = await processExecutor.ExecuteAsync(cmd);
+            return process;
+        }
+
+        private string buildDockerKillCmd()
+        {
+            List<string> cmdList = new List<string>() { Constants.DOCKER_KILL };
+            cmdList.Add(Constants.BrokerLanguageImageMapping[new Tuple<BrokerType, Language>(brokerType, language)]);
+            return string.Join(Constants.SPACE_CHAR, cmdList);
+        }
+
+        private string buildDockerStartCmd() 
```

Comment:
scaling this class would impact with this model. Recommendation would be to isolate the command list creation using a Factory class

Reply:
Changed to factory pattern for shell commands.